src/lib/isLuhn.js

```diff
@@ -0,0 +1,23 @@
+import assertString from './util/assertString';
+
+const luhnArray = [0, 2, 4, 6, 8, 1, 3, 5, 7, 9];
+
+export default function isLuhn(number) {
+  assertString(number);
+  const sanitized = number.replace(/[- ]+/g, '');
+  if (!sanitized) return false;
+  let length = sanitized.length;
+  let bit = 1;
+  let sum = 0;
+  let value;
+
+  while (length) {
+    // eslint-disable-next-line no-plusplus
+    value = parseInt(sanitized.charAt(--length), 10);
+    // eslint-disable-next-line no-bitwise
+    bit ^= 1;
+    sum += bit ? luhnArray[value] : value;
+  }
+
+  return sum % 10 === 0;
+}
```

Comment:
Isn't this the same algorithm used for `isCreditCard`?

Reply:
Same algorithm but in credit card you definer pretty world usage of Luhn, such as **American Express** or etc.  But form example in Iran some Card using this algorithm is not valid Credit Card but also Valid Luhn Number.
For example `5022291047705718` is for https://en.wikipedia.org/wiki/Bank_Pasargad. 
Even you can creat your own Luhn number for any use case.

Credit cards also need to be updated for special countries or baniking, but out of this PR scope.

Reply:
Should we use this algorithm in`isCreditCard` to avoid code repetition?

Reply:
If you need to validate **American Express** no, it's important to validate known luhn algorithm for **American Express**, But generally validation is Luhn is required. Seems this validation required after that **American Express** could be depend on isLuhn and also special prefix of this.

Reply:
@tux-tn 
isLuhn is algorithm verification (as you can see the [popularity on npm](https://www.npmjs.com/search?q=keywords:luhn)), isCreaditCard might be depend on isLuhn. But isLuhn is essential.

Chaning **isCreaditCard** for using(or not) **isLuhn** out of scope of this PR. 

Could you explain, which part need review?